File: python-sdk-examples/knowledge_graph/kg_question.py
import os
import dotenv
from writerai import Writer

dotenv.load_dotenv()

client = Writer()

response = client.graphs.question(
    graph_ids=[os.getenv("GRAPH_ID")],
    question="Which products contain food coloring?",
    stream=True,
    subqueries=True,
)
answer = ""
for chunk in response:
    answer += chunk
    
print(answer)

File: node-sdk-examples/knowledge-graph/kg-question.js
const Writer = require('writer-sdk');
const dotenv = require('dotenv');

dotenv.config();

const client = new Writer({
    apiKey: process.env.WRITER_API_KEY,
});

async function main() {
    try {
      const response = await client.graphs.question({
        graph_ids: [process.env.GRAPH_ID],
        question: 'Which of our products contain food coloring?',
        stream: false,
        subqueries: true,
      });
      console.log(response.answer);
    } catch (error) {
      console.error('Error:', error);
    }
  }

main();